### PR TITLE
22 connection

### PR DESCRIPTION
**Pull request by Herelius**

### CODE REVIEW

**Review by hkamalo — COMMENTED** [review]
Ok for me - wait for the other #22 issue to finish, merge them both before merging to dev

File: app.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 const express = require('express');
 const cors = require('cors');
-const session = require('express-session');
-const MongoStore = require('connect-mongo');
+const expressSession = require('express-session');
```
Comment (edouardmisset) [inline app.js]: Why renaming it expressSession and not keep session ?

File: app.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
+const PgSession = require('connect-pg-simple')(expressSession);
```
Comment (edouardmisset) [inline app.js]: Follow-up from above: is it the right name ? 
Isn't it a connection instead... (postgreConnection maybe ?)

File: db.js [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+  'open',
+  () => {
+    console.log('database connection established successfully');
+  }
+);
 
-module.exports = connection;
+module.exports = {
+  db,
+  prisma,
```
Comment (edouardmisset) [inline db.js:26]: Why do we export Prisma AND db ?
Shouldn't we simply export a connexion to the db like Pierre ?

BTW : Pierre's db.js file is 3 lines :
```javascript
const { PrismaClient } = require('@prisma/client');

const prisma = new PrismaClient();

module.exports = prisma;
```

File: models/ExperimentModel.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const mongoose = require('mongoose');
+// const mongoose = require('mongoose');
```
Comment (edouardmisset) [inline models/ExperimentModel.js:1]: Delete it instead of commenting out 😉 
```suggestion

```

File: models/LocationModel.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-const mongoose = require('mongoose');
+// const mongoose = require('mongoose');
```
Comment (edouardmisset) [inline models/LocationModel.js:1]: Delete it instead of commenting out 😉 
```suggestion

```

File: models/UserModel.js [hunk 6]
```diff
@@ -1,22 +1,7 @@
-const mongoose = require('mongoose');
+// const mongoose = require('mongoose');
```
Comment (edouardmisset) [inline models/UserModel.js]: Delete it instead of commenting out 😉 
```suggestion

```

File: models/UserModel.js [hunk 7]
```diff
@@ -30,10 +15,11 @@ const hashPassword = (plainPassword) =>
 
 // Function to validate the unicity of a user username
```
Comment (edouardmisset) [inline models/UserModel.js:15]:
```suggestion
// Function to validate the unicity of a username
```

File: models/UserModel.js [hunk 8]
```diff
@@ -30,10 +15,11 @@ const hashPassword = (plainPassword) =>
...
 const usernameAlreadyExists = async (username) =>
-  !!(await User.findOne({ username }));
+  !!(await prisma.user.findFirst({ where: { username } }));
```
Comment (edouardmisset) [inline models/UserModel.js:17]: [Prisma's doc](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique)
```suggestion
  !!(await prisma.user.findUnique({ where: { username } }));
```

File: models/UserModel.js [hunk 9]
```diff
@@ -30,10 +15,11 @@ const hashPassword = (plainPassword) =>
...
 
 // Function to return the user with a given username
-const findByUsername = async (username) => User.findOne({ username });
+const findByUsername = async (username) =>
+  prisma.user.findOne({ where: { username } });
```
Comment (edouardmisset) [inline models/UserModel.js]: [Prisma's doc](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique)

```suggestion
await  prisma.user.findUnique({ where: { username } });
```

File: models/UserModel.js [hunk 10]
```diff
@@ -43,14 +29,28 @@ const verifyPassword = (plainPassword, hashedPassword) =>
 const validate = (data) =>
   Joi.object({
     username: Joi.string().max(255).required(),
-    password: Joi.string().min(8).max(100).required(),
+    password: Joi.string().min(1).max(100).required(),
```
Comment (edouardmisset) [inline models/UserModel.js:31]: One seems really short for a minimum password length 😉 

```suggestion
    password: Joi.string().min(4).max(100).required(),
```

File: models/UserModel.js [hunk 11]
```diff
@@ -43,14 +29,28 @@ const verifyPassword = (plainPassword, hashedPassword) =>
...
   }).validate(data, { abortEarly: false }).error;
 
+const findMany = () => prisma.user.findMany();
+
+const create = async ({ username, password }) => {
+  const hashedPassword = await hashPassword(password);
+  return prisma.user.create({
```
Comment (edouardmisset) [inline models/UserModel.js:38]:
```suggestion
await  return prisma.user.create({
```

File: routes/currentUser.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+const currentUserRouter = require('express').Router();
+const asyncHandler = require('express-async-handler');
+const requireCurrentUser = require('../middlewares/requireCurrentUser');
+
+currentUserRouter.get(
+  '/',
+  requireCurrentUser,
+  asyncHandler(async (req, res) => {
+    console.log(req.currentUser);
```
Comment (edouardmisset) [inline routes/currentUser.js]: Why the console log ? dead code ?

```suggestion

```

File: routes/currentUser.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+    res.json(req.currentUser);
```
Comment (edouardmisset) [inline routes/currentUser.js]: Probably need to wrap it in a getSafeAttributes() function like Pierre did.
Also, need to create this function in the model 😉 

```suggestion
    res.json(User.getSafeAttributes(req.currentUser));
```

File: routes/index.js [hunk 14]
```diff
@@ -1,13 +1,15 @@
...
   app.use('/users', usersRouter);
-  app.use('/auth', authRouter);
-  app.use('/locations', locationsRouter);
-  app.use('/sensors', sensorsRouter);
-  app.use('/experiments', experimentsRouter);
+  // app.use('/auth', authRouter);
+  // app.use('/currentUser', currentUserRouter);
+  // app.use('/locations', locationsRouter);
+  // app.use('/sensors', sensorsRouter);
+  // app.use('/experiments', experimentsRouter);
```
Comment (edouardmisset) [inline routes/index.js:14]: Why all this dead code ??

**Review by edouardmisset — CHANGES_REQUESTED** [review]
@Herelius : A few comments to look at

**Pull request closed without merging**